terraform_compliance/steps/steps.py

```diff
@@ -240,20 +240,34 @@ def its_value_must_be_set_by_a_variable(step_obj):
     step_obj.context.stash.property(step_obj.context.search_value).should_match_regex(r'\${var.(.*)}')
 
 
-@then(u'it must not have {proto:ANY} protocol and port {port:ANY} for {cidr:ANY}')
-def it_must_not_have_proto_protocol_and_port_port_for_cidr(step_obj, proto, port, cidr):
+@then(u'it {condition:ANY} have {proto:ANY} protocol and port {port} for {cidr:ANY}')
+def it_condition_have_proto_protocol_and_port_port_for_cidr(step_obj, condition, proto, port, cidr):
     proto = str(proto)
     cidr = str(cidr)
+    ports = port
 
     # In case we have a range
     if '-' in port:
         from_port, to_port = port.split('-')
+    # In case we have comma delimited ports
+    elif ',' in port:
+        print condition
+        assert condition == 'must only', "Comma delimited ports only for must only condition"
+        from_port = to_port = '0'
+        ports = port.split(',')
     else:
         from_port = to_port = port
 
+    if condition == 'must only':
+        condition = True
+    elif condition == 'must not':
+        condition = False
+    else:
+        raise AssertionError("Condition should be 'must only' or 'must not'")
```

Comment:
instead of having this, I think it is better to create another custom type by using `@custom_type` decorator for defining if it is `only` or `not`, on the `@then` definition. That will eliminate few code smell problems.

Additionally having that kind of a path with also enable you to have just ;
```python
condition = condition == 'only'
```
usage.